apps/homework/tasks.py

```diff
@@ -33,37 +34,48 @@ def post_submission(submission_pk):
     )
 
     # Example of url format we're expecting: competition/15595/submission/44798/4aba772a-a6c1-4e6f-a82b-fb9d23193cb6.zip
-
+    if not resp.ok:
+        return
     submission_data = resp.json()['id']
     s3_file_url = resp.json()['url']
 
     with TemporaryFile() as f:
-        parsed_repo_uri = urlparse(submission.github_url)
-        repo_scheme = parsed_repo_uri.scheme
-        repo_loc = parsed_repo_uri.netloc
-        repo_path = parsed_repo_uri.path
-        path_components = [component if component != 'blob' else 'raw' for component in repo_path.split('/')]
-        new_path = ""
-        for index, component in enumerate(path_components):
-            if index != len(path_components) - 1:
-                new_path += component + '/'
-            else:
-                new_path += component
-        repo_url = '{scheme}://{domain}{path}'.format(scheme=repo_scheme, domain=repo_loc, path=new_path)
-        repo_resp = requests.get(repo_url, stream=True)
-        for chunk in repo_resp.iter_content(chunk_size=1024):
-            if chunk:  # filter out keep-alive new chunks
-                f.write(chunk)
-        temp_size = str(f.tell())
-        f.seek(0)
-        storage_resp = requests.put(
-            url=s3_file_url,
-            data=f,
-            headers={
-                "x-ms-blob-type": 'BlockBlob',
-                "Content-Length": temp_size,
-            }
-        )
+        if not data_file:
+            parsed_repo_uri = urlparse(submission.github_url)
+            repo_scheme = parsed_repo_uri.scheme
+            repo_loc = parsed_repo_uri.netloc
+            repo_path = parsed_repo_uri.path
+            path_components = [component if component != 'blob' else 'raw' for component in repo_path.split('/')]
+            new_path = ""
+            for index, component in enumerate(path_components):
+                if index != len(path_components) - 1:
+                    new_path += component + '/'
+                else:
+                    new_path += component
+            repo_url = '{scheme}://{domain}{path}'.format(scheme=repo_scheme, domain=repo_loc, path=new_path)
+            repo_resp = requests.get(repo_url, stream=True)
+            for chunk in repo_resp.iter_content(chunk_size=1024):
+                if chunk:  # filter out keep-alive new chunks
+                    f.write(chunk)
+            temp_size = str(f.tell())
+            f.seek(0)
+            storage_resp = requests.put(
+                url=s3_file_url,
+                data=f,
+                headers={
+                    "x-ms-blob-type": 'BlockBlob',
+                    "Content-Length": temp_size,
+                }
+            )
+        else:
+            storage_resp = requests.put(
```

Comment:
I feel like this is a copy of lines 62-69 above, can we factor this out and reduce DRY violations?

Reply:
👍